share/completions/terraform.fish

```diff
@@ -15,60 +15,193 @@ function __fish_terraform_needs_command
     return 0
 end
 
+function __fish_terraform_using_command
```

Comment:
In this case this should be fine since options alowed before Terraform subcommand don't take any arguments (e.g. `terraform --help plan`). I get your point about Git completions thought.